### PR TITLE
Project-movies Sari-Anne Peltola

### PR DESCRIPTION
**Pull request by sarianne-peltola**
Link to my deployed project: https://distracted-kirch-02aa78.netlify.app/

My reflections on how this week's project turned out: It felt easy! :)

Things I'd like to have clarified or explained in more detail: catch() as an error handler

### CODE REVIEW

File: code/src/pages/ErrorPage.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  return (
+    <div className='error-page'>
+      <h1>
+      <i className="fas fa-bug"></i>
+        Sorry... The page could not be found.</h1>
+      <Link to='/' className='error-link'>
+        Go back to main page
+      </Link>
+    </div>
+  )
```
Comment (Bellsibub) [inline code/src/pages/ErrorPage.js:14]: Very nice error page! 😄

File: code/src/index.css [hunk 2]
```diff
@@ -1,13 +1,334 @@
...
+  height: 1em;
+  border-radius: 50%;
+  position: relative;
+  text-indent: -9999em;
+  -webkit-animation: load5 1.1s infinite ease;
+  animation: load5 1.1s infinite ease;
+  -webkit-transform: translateZ(0);
+  -ms-transform: translateZ(0);
+  transform: translateZ(0);
+}
```
Comment (Bellsibub) [inline code/src/index.css:154]: Very good! 👍 
Sadly the API is so fast I only get to see it for a second 😆

File: code/src/index.css [hunk 3]
```diff
@@ -1,13 +1,334 @@
...
+  top: 30%;
+  left: 10%;
+}
+
+.fa-bug {
+  margin-right: 10px;
+}
+
+.error-link {
+  /* text-decoration: none; */
```
Comment (Bellsibub) [inline code/src/index.css:135]: Commented out code should be removed 😉

File: code/src/components/Loading.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import React from 'react'
+
+export const Loading = () => {
+  return (
+    <div class="loader">Loading...</div>
```
Comment (Bellsibub) [inline code/src/components/Loading.js:5]: ❗❗ with react components we should use `className`
```suggestion
    <div className="loader">Loading...</div>
```

File: code/src/pages/MovieList.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+      setLoading(false);
+    }
+  }, [movieList, loading]);
+
+  return loading ? (
+    <Loading />
+  ) : movieList ? (
+    <section className="movie-list">
+      {movieList.map((movie) => (
+        <MovieListItems {...movie} />
```
Comment (Bellsibub) [inline code/src/pages/MovieList.js:30]: ❗❗ When mapping we should add a key prop to the react element
```suggestion
        <MovieListItems key={movie.id} {...movie} />
```

File: code/src/pages/MovieDetail.js [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+  }, [DETAIL_URL]);
+
+  useEffect(() => {
+    if (movie.success === false) {
+      setErrorPage(true);
+    }
+    if (movie) {
+      setLoading(false);
+    }
+  }, [movie, errorPage, loading]);
```
Comment (Bellsibub) [inline code/src/pages/MovieDetail.js:27]: ❗❗ This structure is causing the fetch to fail at first and then succeed. Cause it calls it twice (I think). I had the same issue in my project and solved it by moving setLoading to the fetch function.
This is the error I get: 
image

**Here is my solution**

```suggestion
  useEffect(() => {
    setLoading(true);
    fetch(DETAIL_URL)
      .then((response) => response.json())
      .then((json) => {
        setMovie(json)
        setLoading(false);
      });
  }, [DETAIL_URL]);

  useEffect(() => {
    if (movie.success === false) {
      setErrorPage(true);
    }
  }, [movie, errorPage]);
```

**Review by Bellsibub — CHANGES_REQUESTED** [review]
Awsome project! 👍 
I really liked your structure and styling 😄 

Sadly I did get some errors when running the website. I therefore have to request changes on this. But other than that the site works perfectly 😍 
I've identified the issues and added comments for each one below. Hopefully they are easy fixes 😉 
(Each issue are marked with ❗❗ )

**Review by KarolCislowski — CHANGES_REQUESTED** [review]
Nice project. @Bellsibub pointed all the little things that need to be fixed, so I will not repeat them here. 

You asked about the `catch()` block: It was created to catch any errors before they destroy the flow of your app and to do something to prevent it from the crash.

**Comment by KarolCislowski** [comment]
Hello, it's your project. Do you remember me? I may be imperfect, but I believe you can still change that before you sit down to your final masterpiece!
image

File: code/src/index.css (hunk 3 above)
Comment (sarianne-peltola) [reply code/src/index.css:135]: Done

File: code/src/components/Loading.js (hunk 4 above)
Comment (sarianne-peltola) [reply code/src/components/Loading.js:5]: Done

File: code/src/pages/MovieList.js (hunk 5 above)
Comment (sarianne-peltola) [reply code/src/pages/MovieList.js:30]: done

File: code/src/pages/MovieDetail.js (hunk 6 above)
Comment (sarianne-peltola) [reply code/src/pages/MovieDetail.js:27]: Done, it worked

**Comment by sarianne-peltola** [comment]
Changes done!

**Review by KarolCislowski — APPROVED** [review]
Nice App 🎉

**Pull request closed without merging**